src/pip/_internal/cli/parser.py

```diff
@@ -197,15 +197,20 @@ def _update_defaults(self, defaults):
             if option is None:
                 continue
 
-            if option.action in ('store_true', 'store_false', 'count'):
+            if option.action in ('store_true', 'store_false'):
```

Comment:
Does this change not mean that
```
    quiet=true
```

is no longer valid in a config file? That would be a breaking change, and not one that I think we should make.

In fact, if this change didn't cause a test to fail (that checked that `quiet=true` worked in a config file) then we should probably add such a test, so that we don't accidentally break that behaviour...

Reply:
I can make both the config-like boolean and int work, and I agree that it's more sensible to do so, since there's no reason to enforce the env var/config option to be strictly int.  I also agree on the tests thing.

Reply:
Not sure why this got marked as resolved (maybe because of a force-push?) I don't see a fix yet, so I've unresolved it. Hope that's OK.

Reply:
Oops, sorry, I meant to make a reply that I'll write the tests somewhere in the near future to notice @deveshks as well.  Regarding the behavior,  `quite=true` is working as of c810ee9.  However, thanks to your checking, I found that the behavior I intended to push is broken.

I'm sorry for resolve the conversation too soon.  For the moment I'll convert this back into a draft PR until I rethink everything out with proper tests to avoid draining others' attention to something that's not ready.

Reply:
No worries, I assumed all was OK, just a glitch. My main worry was that it had been resolved by accident and we'd both forget to check it!